### PR TITLE
docs: Add platform support page

### PR DESCRIPTION
**Pull request by killianmuldoon**
Signed-off-by: killianmuldoon <cilliancapi@gmail.com>

Add the platform support page.

### CODE REVIEW

File: docs/platform-support.rst [hunk 1]
```diff
@@ -19,4 +19,91 @@
...
+   :header-rows: 1
+
+   * - Component
+     - Version
+     - Notes
+   * - Node Feature Discovery
+     - v0.13.2
+     - Optionally deployed. May already be present in the cluster with proper configuration.
+   * - NVIDIA MLNX_OFED driver container
+     - 23.10-0.5.5.0
```
Comment (rollandf) [inline docs/platform-support.rst:84]: Please use vars for component versions. e.g `|mofed-version|`

Need also to add `.. include:: ./common/vars.rst` after `... headings`

File: docs/platform-support.rst [hunk 2]
```diff
@@ -19,4 +19,91 @@
...
+     - -
+   * - IPoIB CNI
+     - v1.1.0
+     - -
+   * - IB Kubernetes
+     - v1.0.2
+     - -
+   * - NV IPAM Plugin
+     - v0.1.1
+     - -
```
Comment (rollandf) [inline docs/platform-support.rst:109]: These is a long list of open source components in the original.
Do we need it? I think that Nadav worked on this list some time ago.

Comment (killianmuldoon) [reply docs/platform-support.rst:109]: I'm not sure which list you're referring to - could you link it?

Comment (rollandf) [reply docs/platform-support.rst:109]: https://docs.nvidia.com/networking/display/kubernetes2410/platform+support#src-2494425313_safe-id-UGxhdGZvcm1TdXBwb3J04oCLLU9wZW5zb3VyY2VfRGVwZW5kZW5jaWVz

File: docs/platform-support.rst [hunk 3]
```diff
@@ -15,8 +15,95 @@
...
+* BlueField-2 NIC Mode
+* BlueField-3 NIC Mode
+
+*************
+Prerequisites
+*************
+
+.. list-table::
+   :header-rows: 1
+   * - Component
```
Comment (rollandf) [inline docs/platform-support.rst:58]: You need an empty line before the rows

File: docs/platform-support.rst [hunk 4]
```diff
@@ -15,8 +15,95 @@
...
+***********************
+Tested Network Adapters
+***********************
+The following network adapters have been tested with the Network Operator:
+
+* ConnectX-6 Dx
+* ConnectX-7
+* BlueField-2 NIC Mode
+* BlueField-3 NIC Mode
+
```
Comment (rollandf) [inline docs/platform-support.rst:51]: Also missing 'Supported ARM Based Platforms'
Make sure to look at 24.1 page.
https://docs.nvidia.com/networking/display/kubernetes2410/platform+support

File: docs/platform-support.rst [hunk 5]
```diff
@@ -15,8 +15,95 @@
...
 .. headings # #, * *, =, -, ^, "
-
+.. include:: ./common/vars.rst
 
 ****************
 Platform Support
-****************
\ No newline at end of file
+****************
+
```
Comment (rollandf) [inline docs/platform-support.rst:23]: Add also a page TOC:
```
.. contents:: On this page
   :depth: 4
   :local:
   :backlinks: none
```

File: docs/platform-support.rst (hunk 2 above)
Comment (killianmuldoon) [reply docs/platform-support.rst:132]: Let me take this as a follow-up. Seems impossible for us to keep this up to date.

**Pull request merged**